v2/sourcedb-to-spanner/terraform/samples/sharded-bulk-migration/main.tf

```diff
@@ -0,0 +1,78 @@
+locals {
+  sharding_config = jsondecode(file(var.common_params.local_sharding_config))
+  dataShards      = local.sharding_config.shardConfigurationBulk.dataShards
+  # Generate individual source configs for each group of data shards based on batch size.
+  source_configs = [
+    for batch_start in range(0, length(local.dataShards), var.common_params.batch_size) : {
+      configType : "dataflow",
+      shardConfigurationBulk : {
+        schemaSource : local.sharding_config.shardConfigurationBulk.schemaSource,
+        dataShards : slice(local.dataShards, batch_start, min(batch_start + var.common_params.batch_size, length(local.dataShards)))
+      }
+    }
+  ]
+
+  working_directory_gcs = "gs://${var.common_params.working_directory_bucket}/${var.common_params.working_directory_prefix}"
+}
+
+resource "random_pet" "job_names" {
+  count = length(local.source_configs)
+}
```

Comment:
Add the `smt-` prefix.
Nit: Rename `job_names` to `prefix` or something similar.

Reply:
With terraform in place, do we want to add smt prefixes? I think we shoud even remove it from the live terraform templates.

Reply:
Terraform is also in under the ambit of "SMT" from a customer perspective. You are thinking `SMT = Harbourbridge`, I am thinking of `SMT = anything that the migration team provides`. The current job names are random and there is no way to filter or aggregate resources created by our tooling in the customer's account - so having a prefix is very helpful.